resdk/resources/base.py

```diff
@@ -87,19 +91,49 @@ def __init__(self, slug=None, id=None,  # pylint: disable=redefined-builtin
 
         self.logger = logging.getLogger(__name__)
 
-    def _update_fields(self, fields):
+    def fields(self):
+        return self.WRITABLE_FIELDS + self.UPDATE_PROTECTED_FIELDS + self.READ_ONLY_FIELDS
+
+    def _update_fields(self, payload):
         """Update resource fields.
 
-        :param fields: Resource fields
-        :type fields: dict
+        :param payload: Resource fields
+        :type payload: dict
 
         """
-        for field_name, field_value in fields.items():
-            setattr(self, field_name, field_value)
+        self._original_values = payload
+        for field_name in self.fields():
+            setattr(self, field_name, payload.get(field_name, None))
 
     def update(self):
         """Update resource fields from the server."""
-        self._update_fields(self.api(self.id).get())
+        response = self.api(self.id).get()
+        self._update_fields(response)
+
+    def save(self):
+        """Save resource to the server."""
+        if self.id:  # update resource
+            payload = {}
+            for field_name in self.WRITABLE_FIELDS:
```

Comment:
We had similar discussion about permissions on resolwe (what happens if user only has permissions for one part of the request) and final decision was that whole request gets rejected. Should we follow similar logic here?